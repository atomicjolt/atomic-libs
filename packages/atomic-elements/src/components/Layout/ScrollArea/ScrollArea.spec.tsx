import { render } from "@testing-library/react";
import { expect, it, describe } from "vitest";
import { ScrollArea } from ".";

describe("ScrollArea", () => {
  describe("Snapshots", () => {
    it("should match snapshot", () => {
      const res = render(<ScrollArea />);
      expect(res).toMatchSnapshot();
    });
  });
});
