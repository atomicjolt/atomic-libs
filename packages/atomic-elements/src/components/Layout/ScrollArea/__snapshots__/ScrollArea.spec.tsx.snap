// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ScrollArea > Snapshots > should match snapshot with basic structure 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-FEMpB bspVW aje-scrollarea"
      >
        <div
          class="sc-gjcoXW bIfMbm aje-scrollarea__viewport"
          style="height: 200px; width: 300px;"
        >
          <div
            style="height: 400px; width: 600px;"
          >
            <p>
              Content that overflows both directions
            </p>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-FEMpB bspVW aje-scrollarea"
    >
      <div
        class="sc-gjcoXW bIfMbm aje-scrollarea__viewport"
        style="height: 200px; width: 300px;"
      >
        <div
          style="height: 400px; width: 600px;"
        >
          <p>
            Content that overflows both directions
          </p>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ScrollArea > Snapshots > should match snapshot with both scrollbars 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-FEMpB bspVW aje-scrollarea"
      >
        <div
          class="sc-gjcoXW bIfMbm aje-scrollarea__viewport"
          style="height: 200px; width: 300px;"
        >
          <div
            style="height: 400px; width: 600px;"
          >
            <p>
              Content that overflows in both directions
            </p>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-FEMpB bspVW aje-scrollarea"
    >
      <div
        class="sc-gjcoXW bIfMbm aje-scrollarea__viewport"
        style="height: 200px; width: 300px;"
      >
        <div
          style="height: 400px; width: 600px;"
        >
          <p>
            Content that overflows in both directions
          </p>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ScrollArea > Snapshots > should match snapshot with custom root props 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-FEMpB bspVW aje-scrollarea custom-scroll-area"
      >
        <div
          class="sc-gjcoXW bIfMbm aje-scrollarea__viewport"
        >
          <div
            style="height: 400px; width: 600px;"
          >
            <p>
              Content with custom root props
            </p>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-FEMpB bspVW aje-scrollarea custom-scroll-area"
    >
      <div
        class="sc-gjcoXW bIfMbm aje-scrollarea__viewport"
      >
        <div
          style="height: 400px; width: 600px;"
        >
          <p>
            Content with custom root props
          </p>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ScrollArea > Snapshots > should match snapshot with horizontal scrollbar 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-FEMpB bspVW aje-scrollarea"
      >
        <div
          class="sc-gjcoXW bIfMbm aje-scrollarea__viewport"
          style="height: 200px; width: 300px;"
        >
          <div
            style="height: 100px; width: 600px;"
          >
            <p>
              Content that overflows horizontally
            </p>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-FEMpB bspVW aje-scrollarea"
    >
      <div
        class="sc-gjcoXW bIfMbm aje-scrollarea__viewport"
        style="height: 200px; width: 300px;"
      >
        <div
          style="height: 100px; width: 600px;"
        >
          <p>
            Content that overflows horizontally
          </p>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ScrollArea > Snapshots > should match snapshot with scrollbar props 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-FEMpB bspVW aje-scrollarea"
      >
        <div
          class="sc-gjcoXW bIfMbm aje-scrollarea__viewport"
          style="height: 200px; width: 300px;"
        >
          <div
            style="height: 400px; width: 600px;"
          >
            <p>
              Content with scrollbar props
            </p>
          </div>
        </div>
        <div
          class="sc-dycHyt bIzKlf aje-flex aje-scrollarea__scrollbar"
          style="height: 0px; visibility: hidden;"
        >
          <button
            aria-label="Scroll retreat vertically"
            class="sc-gNZgCX sc-kfeOyU cHjvIr huOvKw aje-btn aje-icon-btn aje-btn--scrollbar is-medium"
            data-icon-button="true"
            tabindex="0"
            type="button"
          >
            <i
              aria-hidden="true"
              class="sc-jbAkgO jwsDCd aje-icon material-icons is-medium"
            >
              arrow_drop_up
            </i>
          </button>
          <div
            class="sc-dVBluf fiQAHo"
            data-active="false"
            data-orientation="vertical"
          >
            <div
              aria-busy="false"
              aria-orientation="vertical"
              aria-valuemax="0"
              aria-valuemin="0"
              aria-valuenow="0"
              class="sc-fLDLck jHHghy"
              data-active="false"
              role="scrollbar"
              style="height: 100%; top: 0%;"
              tabindex="0"
            />
          </div>
          <button
            aria-label="Scroll advance vertically"
            class="sc-gNZgCX sc-kfeOyU cHjvIr huOvKw aje-btn aje-icon-btn aje-btn--scrollbar is-medium"
            data-icon-button="true"
            tabindex="0"
            type="button"
          >
            <i
              aria-hidden="true"
              class="sc-jbAkgO jwsDCd aje-icon material-icons is-medium"
            >
              arrow_drop_down
            </i>
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-FEMpB bspVW aje-scrollarea"
    >
      <div
        class="sc-gjcoXW bIfMbm aje-scrollarea__viewport"
        style="height: 200px; width: 300px;"
      >
        <div
          style="height: 400px; width: 600px;"
        >
          <p>
            Content with scrollbar props
          </p>
        </div>
      </div>
      <div
        class="sc-dycHyt bIzKlf aje-flex aje-scrollarea__scrollbar"
        style="height: 0px; visibility: hidden;"
      >
        <button
          aria-label="Scroll retreat vertically"
          class="sc-gNZgCX sc-kfeOyU cHjvIr huOvKw aje-btn aje-icon-btn aje-btn--scrollbar is-medium"
          data-icon-button="true"
          tabindex="0"
          type="button"
        >
          <i
            aria-hidden="true"
            class="sc-jbAkgO jwsDCd aje-icon material-icons is-medium"
          >
            arrow_drop_up
          </i>
        </button>
        <div
          class="sc-dVBluf fiQAHo"
          data-active="false"
          data-orientation="vertical"
        >
          <div
            aria-busy="false"
            aria-orientation="vertical"
            aria-valuemax="0"
            aria-valuemin="0"
            aria-valuenow="0"
            class="sc-fLDLck jHHghy"
            data-active="false"
            role="scrollbar"
            style="height: 100%; top: 0%;"
            tabindex="0"
          />
        </div>
        <button
          aria-label="Scroll advance vertically"
          class="sc-gNZgCX sc-kfeOyU cHjvIr huOvKw aje-btn aje-icon-btn aje-btn--scrollbar is-medium"
          data-icon-button="true"
          tabindex="0"
          type="button"
        >
          <i
            aria-hidden="true"
            class="sc-jbAkgO jwsDCd aje-icon material-icons is-medium"
          >
            arrow_drop_down
          </i>
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ScrollArea > Snapshots > should match snapshot with vertical scrollbar 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-FEMpB bspVW aje-scrollarea"
      >
        <div
          class="sc-gjcoXW bIfMbm aje-scrollarea__viewport"
          style="height: 200px; width: 300px;"
        >
          <div
            style="height: 400px; width: 200px;"
          >
            <p>
              Content that overflows vertically
            </p>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-FEMpB bspVW aje-scrollarea"
    >
      <div
        class="sc-gjcoXW bIfMbm aje-scrollarea__viewport"
        style="height: 200px; width: 300px;"
      >
        <div
          style="height: 400px; width: 200px;"
        >
          <p>
            Content that overflows vertically
          </p>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
